spec/controllers/links_controller_spec.rb

```diff
@@ -37,90 +34,89 @@
   end
 
   describe 'POST #create' do
-    before(:each) { @link_count = Link.count }
+    let(:link) { build(:link, user: user) }
 
     context "when logged in" do
       context "when vanity string is blank" do
         it "generates a vanity string" do
-          link = build(:link, user: user)
-          post :create,
-               link: {
-                 actual: link.actual
-               }, format: :js
-          expect(Link.count).to eq(@link_count + 1)
+          expect do
+            post :create, link: { actual: link.actual }, format: :js
+          end.to change(Link, :count).by(1)
```

Comment:
Refactor spec: replace use of instance variable `@link_count` by checking change in count.